src/test/kotlin/org/jmailen/gradle/kotlinter/functional/EditorConfigTest.kt

```diff
@@ -0,0 +1,93 @@
+package org.jmailen.gradle.kotlinter.functional
+
+import groovy.util.GroovyTestCase.assertEquals
+import java.io.File
+import org.gradle.testkit.runner.TaskOutcome
+import org.intellij.lang.annotations.Language
+import org.jmailen.gradle.kotlinter.functional.utils.kotlinClass
+import org.jmailen.gradle.kotlinter.functional.utils.resolve
+import org.jmailen.gradle.kotlinter.functional.utils.settingsFile
+import org.junit.Before
+import org.junit.Test
+
+internal class EditorConfigTest : WithGradleTest.Kotlin() {
+
+    lateinit var projectRoot: File
+
+    @Before
+    fun setUp() {
+        projectRoot = testProjectDir.root.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                @Language("groovy")
+                val buildScript = """
+                plugins {
+                    id 'kotlin'
+                    id 'org.jmailen.kotlinter'
+                }
+                
+            """.trimIndent()
+                writeText(buildScript)
+            }
+        }
+    }
+
+    @Test
+    fun `lintTask uses default indentation if editorconfig absent`() {
+        projectRoot.resolve("src/main/kotlin/FourSpacesByDefault.kt") {
+            writeText(
+                """ |
+                    |    object FourSpacesByDefault
+                    |        
+                """.trimMargin()
+            )
+        }
+
+        build("lintKotlin").apply {
+            assertEquals(TaskOutcome.SUCCESS, task(":lintKotlinMain")?.outcome)
+        }
+    }
+
+    @Test
+    fun `plugin respects indentSize set in editorconfig`() {
```

Comment:
What about the test case that when both kotlinter.indentSize is specified and there's an .editorconfig with the param, the extension wins? Based on the code I expect this to work of course.

Reply:
Done ✅ Added precedence checks for both `disabledRules` and `indentSize`